--- origsrc/libnice-0.0.6/gst/Makefile.am	2009-03-05 14:52:49.000000000 -0600
+++ src/libnice-0.0.6/gst/Makefile.am	2009-04-26 15:14:45.311163200 -0500
@@ -32,4 +32,4 @@ libgstnice_la_SOURCES = \
 
 libgstnice_la_LIBADD = $(COMMON_LDADD)
 
-libgstnice_la_LDFLAGS = -module -avoid-version
+libgstnice_la_LDFLAGS = -module -avoid-version -no-undefined
--- origsrc/libnice-0.0.6/nice/Makefile.am	2009-03-09 15:34:55.000000000 -0500
+++ src/libnice-0.0.6/nice/Makefile.am	2009-04-26 15:15:27.889288200 -0500
@@ -21,6 +21,7 @@ libnice_la_LIBADD = \
 
 libnice_la_LDFLAGS = \
 	-export-symbols $(srcdir)/libnice.sym \
+	-no-undefined \
 	$(LIBNICE_LT_LDFLAGS)
 
 
