--- origsrc/libnice-0.1.8/agent/Makefile.am	2014-10-09 14:28:21.000000000 -0500
+++ src/libnice-0.1.8/agent/Makefile.am	2015-01-06 19:10:43.501634600 -0600
@@ -78,33 +78,3 @@ pkginclude_HEADERS = \
 if WINDOWS
   libagent_la_LIBADD += -liphlpapi -lws2_32
 endif
-
-#
-# GObject introspection
-#
-# We need --accept-unprefixed because of PseudoTcp and TurnServer.
-#
--include $(INTROSPECTION_MAKEFILE)
-INTROSPECTION_GIRS =
-INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir) --warn-all --accept-unprefixed
-INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
-
-if HAVE_INTROSPECTION
-introspection_sources = $(pkginclude_HEADERS)
-
-Nice-0.1.gir: libagent.la
-Nice_0_1_gir_INCLUDES = GObject-2.0 Gio-2.0
-Nice_0_1_gir_EXPORT_PACKAGES = nice
-Nice_0_1_gir_CFLAGS = $(AM_CFLAGS)
-Nice_0_1_gir_LIBS = libagent.la
-Nice_0_1_gir_FILES = $(introspection_sources)
-INTROSPECTION_GIRS += Nice-0.1.gir
-
-girdir = $(datadir)/gir-1.0
-gir_DATA = $(INTROSPECTION_GIRS)
-
-typelibdir = $(libdir)/girepository-1.0
-typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
-
-CLEANFILES += $(gir_DATA) $(typelib_DATA)
-endif
--- origsrc/libnice-0.1.8/nice/Makefile.am	2015-01-06 18:55:30.958256300 -0600
+++ src/libnice-0.1.8/nice/Makefile.am	2015-01-06 22:21:05.465539800 -0600
@@ -63,3 +63,39 @@ pkginclude_HEADERS = nice.h
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = nice.pc
 
+#
+# GObject introspection
+#
+# We need --accept-unprefixed because of PseudoTcp and TurnServer.
+#
+-include $(INTROSPECTION_MAKEFILE)
+INTROSPECTION_GIRS =
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir) --warn-all --accept-unprefixed
+INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
+
+if HAVE_INTROSPECTION
+introspection_sources = \
+	$(top_srcdir)/agent/agent.h \
+	$(top_srcdir)/agent/candidate.h \
+	$(top_srcdir)/agent/debug.h \
+	$(top_srcdir)/agent/address.h \
+	$(top_srcdir)/agent/interfaces.h \
+	$(top_srcdir)/agent/pseudotcp.h \
+	$(NULL)
+
+Nice-0.1.gir: libnice.la
+Nice_0_1_gir_INCLUDES = GObject-2.0 Gio-2.0
+Nice_0_1_gir_EXPORT_PACKAGES = nice
+Nice_0_1_gir_CFLAGS = $(AM_CFLAGS)
+Nice_0_1_gir_LIBS = libnice.la
+Nice_0_1_gir_FILES = $(introspection_sources)
+INTROSPECTION_GIRS += Nice-0.1.gir
+
+girdir = $(datadir)/gir-1.0
+gir_DATA = $(INTROSPECTION_GIRS)
+
+typelibdir = $(libdir)/girepository-1.0
+typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
+
+CLEANFILES += $(gir_DATA) $(typelib_DATA)
+endif
--- origsrc/libnice-0.1.8/nice/libnice.sym	2014-10-09 15:26:20.000000000 -0500
+++ src/libnice-0.1.8/nice/libnice.sym	2015-01-06 22:23:39.767633700 -0600
@@ -58,6 +58,7 @@ nice_agent_set_stream_name
 nice_agent_set_stream_tos
 nice_candidate_copy
 nice_candidate_free
+nice_candidate_get_type
 nice_candidate_new
 nice_component_state_to_string
 nice_debug_disable
@@ -73,6 +74,7 @@ pseudo_tcp_socket_close
 pseudo_tcp_socket_connect
 pseudo_tcp_socket_get_error
 pseudo_tcp_socket_get_next_clock
+pseudo_tcp_socket_get_type
 pseudo_tcp_socket_new
 pseudo_tcp_socket_notify_clock
 pseudo_tcp_socket_notify_mtu
